### PR TITLE
Disposal Bins improvements

### LINKED ISSUE #6568: Disposal bins not initialising correctly
Disposal bins don't seem to be initialising correctly:
off
While the expected sprite state is:
ready
~~In addition, disposal bins spawned in without a disposal pipe terminal underneath are spawning as immediately above, and not rotated, as depicted here:
uninstalled
{spawning over a disposal pipe terminal is as expected)~~

### PR DESCRIPTION
**Pull request by MaxIsJoe**
part of #4654
closes #6568

# Changelog:

CL: [New] Disposal bins are now power reliant and require connection to APCs to function.
CL: [New] Disposal bins will consume 1500 extra watts while recharging.
CL: [New] You can now dump the contents of trash bags directly into disposal bins by simply clicking on them.
CL: [New] Added a new scene construction tool for mappers and designers to automatically link devices to nearby APCs during runtime.
CL: [New] Added the ability for mappers and designers to assign events to devices getting connected to APCs from the unity inspector.
CL: [Improvement] Added feedback text for when disposal bins flush and recharge.
CL: [Improvement] Added hover-tooltip interactions text for disposal bins.
CL: [Fix] Fixed disposal bin sprites not showing their correct states when the round starts.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
tests are failing on upstream

**Comment by corp-0** [comment]
are you sure? I see stations missing APC references which wasn't happening before

**Comment by corp-0** [comment]
image
seems related to this PR

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 1]
```diff
@@ -172,9 +172,9 @@ public NetworkIdentity GetMasterNetID()
 	/// </summary>
 	/// <param name="cataloguePage">Index as defined via the inspector on the object.</param>
 	/// <param name="networked">Whether this change should be sent to clients, if server.</param>
-	public void ChangeSprite(int cataloguePage, bool networked = true)
+	public void ChangeSprite(int cataloguePage, bool networked = true, bool forceUpdate = false)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:175]: I told you what the issue was is that, 
The state of the sprite being visible 
is a different
 state from the Sprite being set

**Comment by MaxIsJoe** [comment]
> are you sure? I see stations missing APC references which wasn't happening before

ah, i completely forgot that it only checks for it during runtime. I should update tests to take this new thing into account.

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:175]: that's some outdated code that has been slipped

**Comment by MaxIsJoe** [comment]
hewwo?

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:175]: still here

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:175]: Nope

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:175]: image

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:175]: image

???????????????????????????????????????????????????????????????????????????????????

**Review by corp-0 — APPROVED** [review]

**Pull request merged**